meadow-client/src/Marlowe/Blockly.purs

```diff
@@ -0,0 +1,1144 @@
+
+divValue :: BlockDefinition
+divValue =
+  BlockDefinition
+    $ merge
+        { type: show DivValueType
+        , message0: "%1 / %2 with default %3"
+        , args0:
+          [ Value {name: "value1", check: "value", align: Right}
+          , Value {name: "value2", check: "value", align: Right}
+          , Value {name: "value3", check: "value", align: Right}
+          ]
+        , colour: "135"
+        , output: Just "value"
+        , inputsInline: Just true
+        } defaultBlockDefinition
+
+modValue :: BlockDefinition
+modValue =
+  BlockDefinition
+    $ merge
+        { type: show ModValueType
+        , message0: "%1 % %2 with default %3"
+        , args0:
+          [ Value {name: "value1", check: "value", align: Right}
+          , Value {name: "value2", check: "value", align: Right}
+          , Value {name: "value3", check: "value", align: Right}
+          ]
+        , colour: "135"
+        , output: Just "value"
+        , inputsInline: Just true
+        } defaultBlockDefinition
+
+fromChoiceValue :: BlockDefinition
+fromChoiceValue =
+  BlockDefinition
+    $ merge
+        { type: show FromChoiceValueType
+        , message0: "use value of choice with id: %1 chosen by participant with id: %2 if no choice was made use: %3"
+        , args0:
+          [ Number {name: "choice_id", value: 1.0, min: Just 1.0, max: Nothing, precision: Nothing}
+          , Number {name: "person_id", value: 1.0, min: Just 1.0, max: Nothing, precision: Nothing}
+          , Value {name: "value", check: "value", align: Right}
+          ]
+        , colour: "135"
+        , output: Just "value"
+        , inputsInline: Just true
+        } defaultBlockDefinition
+
+fromOracleValue :: BlockDefinition
+fromOracleValue =
+  BlockDefinition
+    $ merge
+        { type: show FromOracleValueType
+        , message0: "use value of oracle with id: %1 if no oracle exists use: %2"
+        , args0:
+          [ Number {name: "oracle_id", value: 1.0, min: Just 1.0, max: Nothing, precision: Nothing}
+          , Value {name: "value", check: "value", align: Right}
+          ]
+        , colour: "135"
+        , output: Just "value"
+        , inputsInline: Just true
+        } defaultBlockDefinition
+
+toolbox :: forall a b. HTML a b
+toolbox =
+  xml [id_ "blocklyToolbox", style "display:none"]
+    [ category [name "Contracts", colour "0"]
+        [ block [blockType (unwrap nullContract).type] []
+        , block [blockType (unwrap commitContract).type] []
+        , block [blockType (unwrap payContract).type] []
+        , block [blockType (unwrap bothContract).type] []
+        , block [blockType (unwrap choiceContract).type] []
+        , block [blockType (unwrap whenContract).type] []
+        , block [blockType (unwrap whileContract).type] []
+        , block [blockType (unwrap scaleContract).type] []
+        , block [blockType (unwrap letContract).type] []
+        , block [blockType (unwrap useContract).type] []
+        ]
```

Comment:
Assuming my other comments work out, this becomes:

```
  category [name "Contracts", colour "0"]
    (block <<< pure << show <<< ContractBlock <$> bottom..top)
```

...I think. 🤔

Reply:
done